### PR TITLE
RtpsRelay cleanup

### PR DESCRIPTION
**Pull request by mitza-oci**
* Removed unused code and macros
* Moved some large inline functions to cpp files
* Used C++11 features for clarity

### CODE REVIEW

File: tools/rtpsrelay/GuidPartitionTable.cpp [hunk 1]
```diff
@@ -79,17 +76,107 @@ GuidPartitionTable::Result GuidPartitionTable::insert(const OpenDDS::DCPS::GUID_
...
+    write_relay_partitions(relay_partitions);
+  }
+}
+
+void GuidPartitionTable::lookup(StringSet& partitions, const OpenDDS::DCPS::GUID_t& from) const
+{
+  ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+
+  // Match on the prefix.
+  const auto prefix = make_id(from, OpenDDS::DCPS::ENTITYID_UNKNOWN);
```
Comment (iguessthislldo) [inline tools/rtpsrelay/GuidPartitionTable.cpp:131]:
```suggestion
  const auto prefix = make_unknown_guid(from);
```

File: tools/rtpsrelay/GuidPartitionTable.cpp [hunk 2]
```diff
@@ -79,17 +76,107 @@ GuidPartitionTable::Result GuidPartitionTable::insert(const OpenDDS::DCPS::GUID_
...
+}
+
+void GuidPartitionTable::populate_replay(SpdpReplay& spdp_replay,
+                                         const OpenDDS::DCPS::GUID_t& guid,
+                                         const std::vector<std::string>& to_add) const
+{
+  // The partitions are new for this reader/writer.
+  // Check if they are new for the participant.
+
+  const auto prefix = make_id(guid, OpenDDS::DCPS::ENTITYID_UNKNOWN);
```
Comment (iguessthislldo) [inline tools/rtpsrelay/GuidPartitionTable.cpp:160]:
```suggestion
  const auto prefix = make_unknown_id(guid);
```

File: tools/rtpsrelay/GuidPartitionTable.cpp [hunk 3]
```diff
@@ -79,17 +76,107 @@ GuidPartitionTable::Result GuidPartitionTable::insert(const OpenDDS::DCPS::GUID_
...
+      if (config_.allow_empty_partition() || !part.empty()) {
+        spdp_replay.partitions().push_back(part);
+      }
+      continue;
+    }
+
+    const auto pos2 = pos1->second.lower_bound(prefix);
+
+    if (pos2 == pos1->second.end() ||
+        std::memcmp(pos2->guidPrefix, prefix.guidPrefix, sizeof(prefix.guidPrefix)) != 0) {
```
Comment (iguessthislldo) [inline tools/rtpsrelay/GuidPartitionTable.cpp:174]:
```suggestion
    if (pos2 == pos1->second.end() || equal_guid_prefixes(*pos2, prefix)) {
```

File: tools/rtpsrelay/GuidAddrSet.cpp [hunk 4]
```diff
@@ -7,6 +7,103 @@
...
+                                     bool& ip_now_unused, OpenDDS::DCPS::MonotonicTimePoint& updated_expiration)
+{
+  ACE_INET_Addr addr_only(remote_address.addr);
+  addr_only.set_port_number(0);
+  const auto iter = ip_to_ports.find(addr_only);
+  if (iter == ip_to_ports.end()) {
+    return false;
+  }
+
+  std::map<u_short, OpenDDS::DCPS::MonotonicTimePoint>* port_map = nullptr;
```
Comment (iguessthislldo) [inline tools/rtpsrelay/GuidAddrSet.cpp:60]: Maybe make a `typedef`/`using` of this type to share with `PortSet`? Could also make the switch statement a method in `PortSet`.

File: tools/rtpsrelay/README.rst [hunk 5]
```diff
@@ -2,6 +2,9 @@
 RTPS Relay
 ==========
 
+Note: More up-to-date information about the RtpsRelay is in the OpenDDS Developer's Guide
+see docs/devguide/internet_enabled_rtps.rst
```
Comment (iguessthislldo) [inline tools/rtpsrelay/README.rst:6]:
```suggestion
Note: More up-to-date information about the RtpsRelay is in the OpenDDS Developer's Guide
see ``docs/devguide/internet_enabled_rtps.rst``
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Review by jrw972 — APPROVED** [review]

File: tools/rtpsrelay/GuidPartitionTable.cpp [hunk 6]
```diff
@@ -79,17 +76,106 @@ GuidPartitionTable::Result GuidPartitionTable::insert(const OpenDDS::DCPS::GUID_
       spdp_replay.address(address_);
       spdp_replay.guid(rtps_guid_to_relay_guid(OpenDDS::DCPS::make_id(guid, OpenDDS::DCPS::ENTITYID_PARTICIPANT)));
       if (spdp_replay_writer_->write(spdp_replay, DDS::HANDLE_NIL) != DDS::RETCODE_OK) {
-        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: failed to write Relay Partitions\n")));
+        ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: failed to write Relay Partitions\n"));
```
Comment (sonndinh) [inline tools/rtpsrelay/GuidPartitionTable.cpp:79]:
```suggestion
        ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: GuidPartitionTable::insert failed to write Relay Partitions\n"));
```

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**